### DISCUSSION TITLE
Quarkus-gRpc handling concurrent request

### DISCUSSION THREAD

**Discussion by vasanthmn** [quarkusio/quarkus#29610] [Q&A]
Hi 
 I have created a gRPC service as following.
 
 
 @GrpcService
public class TestListener extends TestService 
{

    @Override
    @Blocking
    @Transactional
    public void onRequest(EventRequest aRequest, StreamObserver<EventResponse> aResponseObserver) { 
	 .............
	 }
	 
}

In onRequest legacy JDBC queries are performed, so this method is blocked with @Blocking. The @Blocking should create a new worker thread to handle request instead of the event loop thread. The works fine if there  a single request comes at time. When there are concurrent request coming from client then only one thread is created and the request is executed one after another. The incoming request are handled in serial and not in parallel. @Blocking should span more than one thread if there are concurrent/parallel request from clients. Am I missing some thing here?

Configurations :
quarkus.grpc.server.instances is not set, so I believe its default to 1. 

Quarkus version: 2.5.0.Final

[a comment by an automated account is not shown]

**Comment by cescoffier** [comment]
Interesting. Did you try with the Mutiny API? 
It might be something wrong with the `@Blocking` handling. It uses `executeBlocking` underneath, and by default, it's sequential. However, in this case, we want concurrent calls.

**Reply by cescoffier** [reply]
Looked at the code, and it seems to be ok. However, 2.5.0 is VERY outdated. Please update.

**Comment by vasanthmn** [comment]
@cescoffier 
Thank you. I will try to update to latest Quarkus and will try again.
